Add Ligolo-ng tunneling tool detection

SecMab opened the pull request:
Detects Ligolo-ng agent and proxy execution for network pivoting
- MITRE ATT&CK: T1572 (Protocol Tunneling)
- Covers renamed binaries via OriginalFileName field
- Addresses detection gap for modern tunneling tools

<!--
Thanks for your contribution. Please make sure to fill the contents of this template with the necessary information to ease and speed up the review process.

!!! PLEASE DO NOT DELETE ANY SECTION, COMMENT OR THE CONTENT OF THE TEMPLATE. !!!
-->

### Summary of the Pull Request

This PR adds detection for Ligolo-ng, a modern tunneling tool used for network pivoting. Currently, SigmaHQ has no coverage for this tool despite its widespread use in penetration testing and adoption by threat actors.

<!--
**Please note that this section is required and must be filled**
A short summary of your pull request.
-->

### Changelog

new: Ligolo-ng Tunneling Tool Execution

<!--
** Don't remove this comment **
You need to add one line for every changed file of the PR and prefix one of the following tags:
new:	<title>
update:	<title> - <optional comment>
fix:	<title> - <optional comment>
remove:	<title> - <optional comment>
chore: for non-detection related changes (e.g. dates/titles) and changes on workflow

e.g.
new: Brute-Force Attacks on Azure Admin Account
update: Suspicious Microsoft Office Child Process - add MSPUB.EXE
fix: Malware User Agent - remove legitimate Firefox UA
chore: workflow - update checkout version
remove: Suspicious Office Execution - deprecated in favour of 8f922766-a1d3-4b57-9966-b27de37fddd2
-->

### Example Log Event

N/A - New detection rule (not a false positive fix)

<!--
Fill this in case of false positive fixes
-->

### Fixed Issues

N/A - New rule submission

<!--
Link the fixed issues here, in case your commit fixes issues with rules or code
-->

### SigmaHQ Rule Creation Conventions

- [x] Followed SigmaHQ naming conventions
- [x] Applied proper MITRE ATT&CK tagging (T1572)
- [x] Used unique UUID v4 identifier
- [x] Set status to 'experimental' for new rule

<!--
- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/)
-->

SecMab committed:
Add Ligolo-ng tunneling tool detection

Detects Ligolo-ng agent and proxy execution for network pivoting
- MITRE ATT&CK: T1572 (Protocol Tunneling)
- Covers renamed binaries via OriginalFileName field
- Addresses detection gap for modern tunneling tools

swachchhanda000 reviewed (changes requested):
Hi @SecMab,

Thank you for your contribution!. keep them coming.

However, Before we start with PR review, please make sure:
1.  Your rule follow the [sigma standard specification ](https://github.com/SigmaHQ/sigma-specification/blob/main/specification/sigma-rules-specification.md). You can also check related rules which are already merged on the repo.
2. All automated checks and actions are passing

PSA: We also appreciate corresponding evtx log for regression test.

frack113 commented on rules/windows/process_creation/proc_creation_win_hktl_ligolo_ng.yml:
```diff
@@ -0,0 +1,34 @@
+    - https://github.com/nicocha30/ligolo-ng
+    - https://attack.mitre.org/techniques/T1572/
+author: MAB
+date: 2025/12/31
```
```suggestion
date: 2025-12-31
```
Check the [specification](https://github.com/SigmaHQ/sigma-specification/blob/main/specification/sigma-rules-specification.md#date) for more information

frack113 reviewed (changes requested):
You have to fix your yaml to pass the workflow:
```
======================
= Linting YAML files =
======================
Error: /windows/process_creation/proc_creation_win_hktl_ligolo_ng.yml:19:13: [error] wrong indentation: expected 14 but found 12 (indentation)
Error: Process completed with exit code 1.
```

SecMab committed:
Apply reviewer feedback - fix date format, YAML indentation, remove MITRE URL

- Fixed date format to ISO 8601 (YYYY-MM-DD)
- Corrected YAML indentation errors
- Removed MITRE ATT&CK URL from references
- Maintained detection logic with proper structure

swachchhanda000 commented on rules/windows/process_creation/proc_creation_win_hktl_ligolo_ng.yml:
```diff
@@ -0,0 +1,34 @@
+    selection_img:
+        Image|endswith:
+            - '\agent.exe'
+            - '\proxy.exe'
```
```suggestion
```
These names are also very generic. When writing a rule about hktl,
we try to use the very specific process names, originalfilename or imphash.

In some cases, we also use command lines that may include combinations of flags or any specific strings which is unique to that specific tool only.

Please have a look at other HKTL or PUA rules on the repo for better understanding and make the changes accordingly.
Thanks

SecMab committed:
Refactor: AND logic + CLI spacing for FP reduction

- Changed from OR to AND logic to reduce false positives
- Added CLI switch spacing following SigmaHQ patterns
- Added T1090.001 MITRE tag for Internal Proxy

SecMab committed:
---

---

swachchhanda000 committed:
fix: logic changes and new dns rule

swachchhanda000 committed:
fix: yamlinting

swachchhanda000 committed:
fix: rules

swachchhanda000 committed:
fix: add missing flag

swachchhanda000 committed:
add regression test

swachchhanda000 committed:
Merge branch 'master' into SecMab-patch-1

swachchhanda000 committed:
fix: trailing whitespace

swachchhanda000 committed:
fix: author

swachchhanda000 committed:
Apply suggestions from code review

swachchhanda000 commented on rules/windows/process_creation/proc_creation_win_hktl_ligolo_ng.yml:
```diff
@@ -0,0 +1,40 @@
+        CommandLine|contains:
+            - '-autocert'
+            - '-laddr'
+            - '-selfcert'
```
This seems like the best option possible, but i think the selection might still be prone to FPS issues? What are your thoughts on this, @nasbench?
